operator/pkg/lib/sidecars/pods/filter.go

```diff
@@ -0,0 +1,139 @@
+package pods
+
+import (
+	"encoding/json"
+	v1 "k8s.io/api/core/v1"
+)
+
+const AnnotationResetWarningKey = "istio.reconciler.kyma-project.io/proxy-reset-warning"
+
+func hasIstioSidecarStatusAnnotation(pod v1.Pod) bool {
+	_, exists := pod.Annotations["sidecar.istio.io/status"]
+	return exists
+}
+
+func isPodReady(pod v1.Pod) bool {
+	isMarkedForDeletion := pod.ObjectMeta.DeletionTimestamp != nil
+	return !isMarkedForDeletion && hasTrueStatusConditions(pod) && isPodRunning(pod)
+}
+
+func hasTrueStatusConditions(pod v1.Pod) bool {
+	for _, condition := range pod.Status.Conditions {
+		if condition.Status != v1.ConditionTrue {
+			return false
+		}
+	}
+	return true
+}
+
+func isPodRunning(pod v1.Pod) bool {
+	if pod.Status.Phase != v1.PodRunning {
+		return false
+	}
+	return true
+}
+
+func getIstioSidecarNamesInPod(pod v1.Pod) []string {
+	type istioStatusStruct struct {
+		Containers []string `json:"containers"`
+	}
+	istioStatus := istioStatusStruct{}
+	err := json.Unmarshal([]byte(pod.Annotations["sidecar.istio.io/status"]), &istioStatus)
+	if err != nil {
+		return []string{}
+	}
+	return istioStatus.Containers
+}
+
+func hasSidecarContainerWithWithDifferentImage(pod v1.Pod, expectedImage SidecarImage) bool {
+	// TODO Why can't we simply assume the pod name to be istio-proxy? Because of istio-init?
```

Comment:
Can someone add any information about this?

Reply:
We can safely assume that container name will be `istio-proxy`